eureka-client/src/main/java/com/netflix/discovery/DiscoveryClient.java

```diff
@@ -196,6 +196,30 @@
 
         @Inject(optional = true)
         private Provider<HealthCheckHandler> healthCheckHandlerProvider;
+
+        public EventBus getEventBus() {
```

Comment:
I understand that these methods are an alternative to DiscoveryClientOptionalArgs. As we use constructor injection everywhere, why not add yet another constructor that accepts the additional, optional values directly without a wrapper?